### ISSUE TITLE
MAINT, TST: Unused C API Functions

### ISSUE THREAD

**Issue by tylerjereddy** [numpy/numpy#12275]
There are many [uncovered / unused NumPy C API functions like `PyArray_Max`](https://codecov.io/gh/numpy/numpy/src/master/numpy/core/src/multiarray/calculation.c#L273) in `calculation.c`, which are effectively circumvented by `NPY_FORWARD_NDARRAY_METHOD("_amax")` style calls in `methods.c`.

The functions seem to be registered in `core/code_generators/numpy_api.py`, and presumably can't simply be removed.

If I'm correct in assuming they can't be removed--should we find a way to test them?

**Comment by charris** [comment]
> If I'm correct in assuming they can't be removed

Yes. Compatibility is the main reason, anyone who currently calls them in extension code would need to rewrite their code if they are removed, and even those who didn't use them would need to recompile to the new interface. I think the best we could do is deprecate them, then leave them alone.

We should probably test all the non-deprecated API functions, not sure what the best way to do that is. Having a Python interfaces to all of them would be handy for that. Is there a list of those functions that do not currently have an interface?

**Comment by tylerjereddy** [comment]
Here's an initial list:

- `PyArray_Max`
- `PyArray_Min`
- `PyArray_Ptp`
- `PyArray_Mean`
- `PyArray_Sum`
- `PyArray_Prod`
- `PyArray_All`
-  `PyArray_Std`
- `__New_PyArray_Std`

I guess it is a pain to probe them without Python interfaces--would either need to make one or use a C testing framework, which probably is just as much work to get going.
